dask_kubernetes/utils.py

```diff
@@ -56,19 +56,34 @@ async def get_external_address_for_scheduler_service(
         nodes = await core_api.list_node()
         host = nodes.items[0].status.addresses[0].address
     elif service.spec.type == "ClusterIP":
-        try:
-            # Try to resolve the service name. If we are inside the cluster this should succeeed.
-            host = f"{service.metadata.name}.{service.metadata.namespace}"
-            socket.getaddrinfo(host, port)
-        except socket.gaierror:
-            # If we are outside it will fail and we need to port forward the service.
+        # Try to resolve the service name. If we are inside the cluster this should succeed.
+        host = f"{service.metadata.name}.{service.metadata.namespace}"
+        success = _is_service_available(
+            host=host, port=port, retries=service_name_resolution_retries
+        )
+
+        # If we are outside it will fail and we need to port forward the service.
+        if not success:
```

Comment:
If `_is_service_available` raises the `socket.gaierror` after retrying then this can stay as it is and we can avoid the temporary `success` variable.